0.preprocess-sites/4.image-and-segmentation-qc.py

```diff
@@ -504,14 +508,48 @@
         )
 
     image_corr_df = (
-        pd.concat(image_corr_list).drop_duplicates(subset="site_location").reset_index()
+        pd.concat(image_corr_list).drop_duplicates(subset="site").reset_index()
+    )
+
+    image_corr_df["min_corr_value"] = (
+        image_corr_df[corr_qc_cols].astype(float).max(skipna=True, axis=1)
     )
 
     for col in corr_qc_cols:
         image_corr_df.loc[(image_corr_df[col] >= correlation_threshold), col] = "pass"
 
+    for col in corr_qc_cols:
+        if len(image_corr_df[col].unique()) == 1:
+            image_corr_df = image_corr_df.drop(col, axis=1)
+
     if len(image_corr_df.index) > 0:
         corr_output_file = pathlib.Path(results_output, "flagged_correlations.csv")
 
         if check_if_write(corr_output_file, force, throw_warning=True):
             image_corr_df.to_csv(corr_output_file)
+
+# By-well visualization of sites with poor correlations
+if len(corr_qc_cols) > 0:
+    image_corr_df = image_corr_df.merge(loc_df, on="site")
```

Comment:
check "site" notation - not sure about `loc_df` but I do remember fixing in d98ded387fe376ea310ae23847ea40758de2c388

Reply:
yep, sorry, another casualty of my use of an older version.